builder/confd/confd.go

```diff
@@ -0,0 +1,95 @@
+// Package confd provides basic Confd support.
+//
+// Right now, this library is highly specific to the needs of the present
+// builder. Because the confd library is not all public, we don't use it directly.
+// Instead, we invoke the CLI.
+package confd
+
+import (
+	"github.com/Masterminds/cookoo"
+	"github.com/Masterminds/cookoo/log"
+	"github.com/Masterminds/cookoo/safely"
+
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+)
+
+// defaultEtcd is the default Etcd host.
+const defaultEtcd = "127.0.0.1:4001"
+
+// RunOnce runs the equivalent of `confd --onetime`.
+//
+// This may run the process repeatedly until either we time out (~20 minutes) or
+// the templates are successfully built.
+//
+// Importantly, this blocks until the run is complete.
+//
+// Params:
+// - node (string): The etcd node to use. (Only etcd is currently supported)
+//
+// Returns:
+// - The []bytes from stdout and stderr when running the program.
+//
+func RunOnce(c cookoo.Context, p *cookoo.Params) (interface{}, cookoo.Interrupt) {
+
+	node := p.Get("node", defaultEtcd).(string)
+
+	dargs := []string{"-onetime", "-node", node, "-log-level", "error"}
+
+	log.Info(c, "Building confd templates. This may take a moment.")
+
+	limit := 1200
+	timeout := time.Second * 3
+	var lasterr error
+	start := time.Now()
+	for i := 0; i < limit; i++ {
+		if out, err := exec.Command("confd", dargs...).CombinedOutput(); err == nil {
+			log.Infof(c, "Templates generated for %s on run %d", node, i)
+			return out, nil
+		} else {
```

Comment:
So in your revised example above, `out` is now scoped to the block of the entire for loop. Originally, `out` is just scoped to the `if/else` block. Since there's a `time.Sleep` call, `out` remains allocated for that entire time, and the GC can't clean it up.

The reason I did it the way I did is so that the GC can clean up `out` and that memory is not allocated for the entire time during this goroutine's inactive period.

Clearly, this is a micro-optimization, and realistically it is unlikely to ever make a difference in the Builder's performance. But my argument remains that it is preferable to code this way than to simply do what `golint` suggests.